src/main/java/seedu/address/logic/parser/ScheduleCommandParser.java

```diff
@@ -2,50 +2,41 @@
 
 import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
 
-import java.time.LocalDateTime;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
-import seedu.address.commons.core.index.Index;
-import seedu.address.logic.commands.ScheduleCommand;
+import seedu.address.logic.commands.schedule.AddScheduleCommand;
+import seedu.address.logic.commands.schedule.ScheduleCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.logic.parser.schedule.AddScheduleCommandParser;
 
 /**
  * Parses input arguments and creates a new DeleteCommand object
  */
 public class ScheduleCommandParser implements Parser<ScheduleCommand> {
-
+    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
     /**
      * Parses the given {@code String} of arguments in the context of the DeleteCommand
      * and returns a DeleteCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
-    public ScheduleCommand parse(String args) throws ParseException {
-        String[] input;
-        Index index;
-        try {
-            input = this.parseDelimitedCommand(args);
-            index = ParserUtil.parseIndex(input[0]);
-        } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ScheduleCommand.MESSAGE_USAGE), pe);
+    public AddScheduleCommand parse(String args) throws ParseException {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(args.trim());
```

Comment:
From my understanding, it seems that the `ScheduleCommandParser` deals with the delete functionality. Would it be better to refactor the class name to `DeleteScheduleCommandParser? - not sure if it would make sense to create an interface or abstract class `ScheduleCommandParser`.

Also I'm not sure if the file directory is showing up properly, but I noticed there is a subdirectory for `schedule/AddScheduleCommandParser.java`, but the `ScheduleCommandParser` is not in the subdirectory. Perhaps we can shift it in?

Reply:
hey yeap! There will be another class `DeleteScheduleCommandParser` when implementing the delete functionality.
So basically `ScheduleCommandParser` acts as a "pre-parser" to `AddScheduleCommandParser`, `DeleteScheduleCommandParser` and `EditScheduleCommandParser` by making sure that only the following 3 types `schedule add`, `schedule edit` and `schedule delete` are accepted and then the arguments are passed into their respective parsers.

Reply:
omg I realised I declared the return type of `parse` wrongly. It should be `ScheduleCommand` not `AddScheduleCommand`. Sorry will fix it now.

Reply:
Ohh, I see! hehe thanks for the explanation! :)